### PR TITLE
Add `.stats()` method

### LINKED ISSUE #170: Include stats in result
I wish I thought of this earlier, but would including some stats in the resolved promise be a good idea? Currently you have to write a lot of boilerplate (see [gulp-imagemin](https://github.com/sindresorhus/gulp-imagemin/blob/master/index.js#L60-L69)) if you want to display some sort of summary after optimzation. What bugs me is this would require *another* major and I'm not to keen on that.

There's also the option to add some sort of "reporter" option or whatever.

@sindresorhus @shinnn

### PR DESCRIPTION
**Pull request by kevva**
Fixes #170.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -50,3 +50,15 @@ test('return original buffer if no plugins are defined', async t => {
...
 });
+
+test('show stats for optimized file', async t => {
+	const buf = await fsP.readFile(path.join(__dirname, 'fixture.jpg'));
+	const data = await m.buffer(buf, {plugins: imageminJpegtran()});
+
+	t.deepEqual(m.stats(buf, data), {
+		difference: 3999,
+		max: 50986,
+		min: 46987,
```
Comment (kevva) [inline test.js]: Not sure about the naming on `max` and `min`, but that's the best I could think of.

File: readme.md [hunk 2]
```diff
@@ -73,6 +73,22 @@ Type: `array`
 
 Array of [plugins](https://www.npmjs.com/browse/keyword/imageminplugin) to use.
 
+### imagemin.stats(max, min)
+
```
Comment (sindresorhus) [inline readme.md]: Need to also document the returned value.

File: index.js [hunk 3]
```diff
@@ -62,3 +62,14 @@ module.exports.buffer = (input, opts) => {
...
 };
+
+module.exports.stats = (max, min) => {
+	const difference = Math.abs(max.length - min.length);
+
+	return {
+		difference,
+		max: max.length,
+		min: min.length,
+		percent: (difference / max.length) * 100
```
Comment (sindresorhus) [inline index.js]: We could also provide a default message implementations can use: https://github.com/sindresorhus/gulp-imagemin/blob/d933de19aaa790b567cf724619aa39b975c71280/index.js#L64-L65

**Comment by sindresorhus** [comment]
// @kevva

**Pull request closed without merging**